cocos/particle-2d/particle-system-2d.ts

```diff
@@ -838,20 +836,33 @@ export class ParticleSystem2D extends UIRenderer {
         }
     }
 
-    protected _flushAssembler (): void {
+    public override destroyRenderData (): void {
+        if (this._simulator.renderData && this._assembler) {
+            this._assembler.removeData(this._simulator.renderData);
+            this._simulator.renderData = null;
+        }
+        super.destroyRenderData();
+    }
+
+    protected override _flushAssembler (): void {
         const assembler = ParticleSystem2D.Assembler.getAssembler(this);
 
         if (this._assembler !== assembler) {
             this._assembler = assembler;
         }
         if (this._assembler && this._assembler.createData) {
-            this._simulator.renderData = this._assembler.createData(this);
-            this._simulator.renderData.particleInitRenderDrawInfo(this.renderEntity); // 确保 renderEntity 和 renderData 都是 simulator 上的
-            this._simulator.initDrawInfo();
+            const simulator = this._simulator;
+            let renderData = simulator.renderData;
+            if (!renderData) {
+                renderData = simulator.renderData = this._assembler.createData(this) as MeshRenderData;
+                simulator.uvFilled = 0;
+                renderData.particleInitRenderDrawInfo(this.renderEntity); // 确保 renderEntity 和 renderData 都是 simulator 上的
```

Comment:
It is better don't use chinese for comments.

Reply:
It was the comment in the old code, not I added in this PR.

Reply:
Yep, i see. But can improve it in this PR.

Reply:
Fixed.